checkov/secrets/scan_git_history.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+import git
+
+from detect_secrets.core import scan
+from git import InvalidGitRepositoryError, GitCommandError, Commit  # type: ignore
+
+if TYPE_CHECKING:
+    from detect_secrets import SecretsCollection
+
+
+def get_commits(root_folder: str) -> list[Commit] | None:
+    try:
+        repo = git.Repo(root_folder)
+    except InvalidGitRepositoryError:
+        logging.error(f"Folder {root_folder} is not a GIT project")
+        return None
+    return list(repo.iter_commits(repo.active_branch, max_count=7))
+
+
+def scan_history(root_folder: str, secrets: SecretsCollection) -> None:
+    commits = get_commits(root_folder)
+    if not commits:
+        return
+    scanned_file_count = 0
+    skipped_file_count = 0
+
+    # we scan the diff between the commit and the next commit - start from the end
+    for previous_commit_idx in range(len(commits) - 1, 0, -1):
+        current_commit_idx = previous_commit_idx - 1
+        current_commit_hash = commits[current_commit_idx].hexsha
+        git_diff = commits[previous_commit_idx].diff(current_commit_hash, create_patch=True)
+
+        for file_diff in git_diff:
+            try:
+                if file_diff.renamed:
+                    logging.warning(f"File was renamed from {file_diff.rename_from} to {file_diff.rename_to}")
```

Comment:
if a file renamed, it cannot also be edited?

Reply:
yes but i will take it in the next ticket.